### PR TITLE
#15 implemented the counts for passed/failed/... tests

### PR DESCRIPTION
**Pull request by Almewty**
Implemented the missing feature referenced in issue #15.
Cleaned up the commit history in comparison with my master branch and implemented the suggestions from CharliePoole

### CODE REVIEW

**Comment by ChrisMaddock** [comment]
Thanks for putting this together @Almewty! 

CI's failed on this one due to a unit test. Looks like it just needs updating - would you mind taking a look?

```
Errors, Failures and Warnings
1) Failed : NUnit.Engine.Drivers.Tests.TestEventAdapterTests.SuiteFinished
  Expected string length 315 but was 312. Strings differ at index 163.
  Expected: "..." duration="12.500000" total="1234" passed="0" failed="0" ..."
  But was:  "..." duration="12.500000" total="0" passed="0" failed="0" inc..."
  --------------------------------------------^
at NUnit.Engine.Drivers.Tests.TestEventAdapterTests.SuiteFinished() in C:\projects\nunit-v2-framework-driver\src\nunit.v2.driver.tests\TestEventAdapterTests.cs:line 107
```

@CharliePoole - are you happy to continue reviewing this? NUnit 2 ain't my area!

**Comment by Almewty** [comment]
Yeah I missed that test.
Just pushed an update for the unit test and CI.

File: src/extension/XmlExtensions.cs [hunk 1]
```diff
@@ -149,12 +149,17 @@ private static void AddResult(this XmlNode parent, TestResult result, bool recur
 
             if (result.Test.IsSuite)
             {
-                // TODO: These values all should be calculated
-                thisNode.AddAttribute("total", result.Test.TestCount.ToString());
-                thisNode.AddAttribute("passed", "0");
-                thisNode.AddAttribute("failed", "0");
-                thisNode.AddAttribute("inconclusive", "0");
-                thisNode.AddAttribute("skipped", "0");
+                var resultDict = CountResultStates(result);
```
Comment (CharliePoole) [inline src/extension/XmlExtensions.cs:152]: Nice idea!

**Review by CharliePoole — APPROVED** [review]
Looks good to me! Thanks for doing this.

**Pull request merged**